src/relations/async_replication.py

```diff
@@ -524,11 +524,11 @@ def _on_async_relation_changed(self, event: RelationChangedEvent) -> None:
         if not self._stop_database(event):
             return
 
-        if not all(
+        if not (self.charm.is_unit_stopped or self._is_following_promoted_cluster()) or not all(
```

Comment:
```suggestion
        if not (self.charm.is_unit_stopped or self._is_following_promoted_cluster()) and not all(
```
I think all units have to fulfil the criteria here

Reply:
This part is hard to reason about 😕 

- I read the [original code](https://github.com/canonical/postgresql-operator/blob/cd6cc0cfc9aa88d99f044ec51cc42e9033bb8e71/src/relations/async_replication.py#L527-L532) as: enter block if any of the units is not (stopped or following).
- I read the [proposed code](https://github.com/canonical/postgresql-operator/blob/becfdf5715facfa2e06d4bb71abd4498105b66a3/src/relations/async_replication.py#L527-L532) as: enter block if current unit is not (stopped or following) `OR` any of the rest are not (stopped or following).

The uppercase `OR` is the one you are proposing to change to an `AND`, making it impossible for the code to enter the block unless current unit AND at least one of the rest are not stopped / following.

Reply:
I had the same understanding as @sinclert-canonical here.

Reply:
Hey @dragomirp, did you have time to revisit this topic?